### PR TITLE
Merge Offline Database into Main

### PR DESCRIPTION
**Pull request by Lihemen**
# PR Template

## Description

This PR introduces the following features
1. Rollbar for error reporting
2. Sentry for crashlytics and session replays
3. RxDB as the offline database for an offline-syncable PWA
4. Hooks for performing CRUD operations with the DB (syncable with a backend API)

This PR also fixes the following issues
1. Deletes the package-lock
2. Ensures lint staged runs irrespective of OS
3. All linting and formatting issues in the project

- [x] Bugfix
- [x] New feature
- [x] Refactor
- [ ] Documentation
- [ ] Test

## Related Issues

Please link to the issues this pull request addresses (if any):

- Fixes #(issue number)

## Screenshots (if applicable)

Add any relevant screenshots or GIFs to help visualize the changes (if UI/UX related).

## Checklist

Before submitting, make sure your PR meets these requirements:

- [x] I have tested my changes locally and verified that they are working as intended.
- [x] I have written tests for all new functionality (if applicable).
- [x] I have updated the documentation where necessary.
- [x] I have followed the coding style of this project.
- [ ] My PR has been reviewed by at least one peer (if applicable).
- [ ] This PR is ready for review and merge.

## Additional Context

Any other information or thoughts relevant to this PR. Consider mentioning team members for visibility.

### CODE REVIEW

**Review by Jessica-Uriostegui — CHANGES_REQUESTED** [review]

File: src/features/goal-form/index.tsx
[the remark this answers is not in the document]
Comment (Lihemen) [reply src/features/goal-form/index.tsx:1]: It’s an example of how to sync with and use the database

[the remark this answers is not in the document]
Comment (Lihemen) [reply src/features/goal-form/index.tsx:1]: It’s an example of how to write to and sync with the database.

File: src/features/navigation/mobile/top.navigation.tsx
Comment (Jessica-Uriostegui) [inline src/features/navigation/mobile/top.navigation.tsx:1]: So why does this button lead to GoalForm if its just a sample ?

**Review by Jessica-Uriostegui — CHANGES_REQUESTED** [review]
On top navigation Add button is attached to goalform which was the name of my code page that you changed. so this goes to the page that  you called "sample page" sample page for what? also why was the name changed? why dont you change the name to your page into sample page?  why was the space added to goalform

File: .husky/pre-commit
[the remark this answers is not in the document]
Comment (Lihemen) [reply .husky/pre-commit:1]: This solves for that.

File: src/pages/goals/new/index.tsx
[the remark this answers is not in the document]
Comment (Lihemen) [reply src/pages/goals/new/index.tsx:1]: Goal form is a generic name that doens't encapsulate anything.

NewGoal is more specific.

Also, all exports are named exports, using GoalForm for a component that connotes a new goal is not ideal.

File: src/pages/goals/edit/index.tsx
[the remark this answers is not in the document]
Comment (Lihemen) [reply src/pages/goals/edit/index.tsx:1]: That is the linter at work.

File: src/features/navigation/mobile/top.navigation.tsx
Comment (Lihemen) [reply src/features/navigation/mobile/top.navigation.tsx:1]: Lead to what?
It leads to a path /goals/new. Have you checked the component rendered at that path?

**Comment by Lihemen** [comment]
> On top navigation Add button is attached to goalform which was the name of my code page that you changed. so this goes to the page that you called "sample page" sample page for what? also why was the name changed? why dont you change the name to your page into sample page? why was the space added to goalform



@Jessica-Uriostegui the button goes to the path /goals/new. I'm trying hard to see how this is difficult to understand. This is a non-issue.

**Pull request closed without merging**